cloud/endagaweb/tasks.py

```diff
@@ -439,3 +439,124 @@ def req_bts_log(self, obj, retry_delay=60*10, max_retries=432):
+                    subscriber.save()
+                    print "Updating subscriber(%s) state to 'First Expired'" % (
+                        subscriber.imsi,)
+            elif today > recycle:
+                # Let deactivation of subscriber be handled by
+                # vacuum_inactive_subscribers
+                # Do nothing if it's already recycle
+                if current_state != 'recycle':
+                    subscriber.state = 'recycle'
+                    subscriber.save()
+                    print "Updating subscriber(%s) state to 'Recycle'" % (
+                        subscriber.imsi,)
+            else:
+                if current_state != 'expired':
+                    subscriber.state = 'expired'
+                    subscriber.save()
+                    print "Updating subscriber(%s) state to 'Expired'" % (
+                        subscriber.imsi,)
+
+
+@app.task(bind=True)
+def validity_expiry_sms(self, days=7):
+    """Sends SMS to the number whose validity is:
+     about to get expire, 
+     if expired (i.e 1st expired), or
+     if the number is in grace period and is about to recycle.
+
+     Args:
+         days: Days prior (state change) which the SMS is sent to Subscriber.
+     Runs as everyday task managed by celerybeat.
+     """
+    today = django.utils.timezone.datetime.now().date()
+    for subscriber in Subscriber.objects.iterator():
+        # Do nothing if subscriber vacuuming is disabled for the network.
+        if not subscriber.network.sub_vacuum_enabled:
+            continue
+        try:
+            number = subscriber.number_set.all()[0]
+            subscriber_validity = number.valid_through
+            # In case where number has no validity
+            if subscriber_validity is None:
+                print '%s has no validity' % (subscriber.imsi,)
+                continue
+        except IndexError:
+            print 'No number attached to subscriber %s' % (subscriber.imsi,)
+            continue
+
+        subscriber_validity = subscriber_validity.date()
+        inactive_period = subscriber.network.sub_vacuum_inactive_days
+        grace_period = subscriber.network.sub_vacuum_grace_days
+
+        prior_first_expire = subscriber_validity + datetime.timedelta(
+            days=inactive_period) - datetime.timedelta(days=days)
+
+        prior_recycle = prior_first_expire + datetime.timedelta(
+            days=grace_period)
+
+        # Prior to expiry state (one on last day and before defined days)
+        if subscriber_validity > today and (
+                            (subscriber_validity - datetime.timedelta(
+                                days=days)
+                             ) == today or today == (
+                                subscriber_validity - datetime.timedelta(
+                                days=1)) or today == subscriber_validity):
+            body = 'Your validity is about to get expired on %s , Please ' \
```

Comment:
this needs be translated so this will need to live on the client. you can construct a key/short phrase that you send that the client translates with local i18n